### PR TITLE
Cypress backports

### PR DESCRIPTION
**Pull request by Kelketek**
Backports and final adjustments

### CODE REVIEW

**Comment by Kelketek** [comment]
@bradenmacdonald @smarnach This requires your review upon your first opportunity.

CircleCI is not able to install all of the prereqs without having some connection error along the way and dying at a different one each time.

**Comment by smarnach** [comment]
@Kelketek I fixed CircleCI in [this commit](https://github.com/edx-solutions/edx-platform/commit/8426b68eec7f1bf754db6aab2370acc514b475c9), which I pushed to [this PR](https://github.com/edx-solutions/edx-platform/pull/485), but it unfortunately got eaten by some force push before merging.  The exact error CircleCI is reporting is fixed by this commit.  Could you try reapplying it?

**Comment by smarnach** [comment]
The code changes look good to me.  Will the PEP8 fixes also be applied upstream, or have they already been applied upstream?  Otherwise, they might cause the solutions fork to unnecessarily diverge from upstream.

**Comment by Kelketek** [comment]
@smarnach The PEP8 check those fixes are for is all-encompassing both upstream and here. So the fixes should only be near things we've fiddled with anyway.

**Comment by Kelketek** [comment]
@smarnach @bradenmacdonald Fixed the gradebook test failures. The build is running again. At the moment this PR contains a few things that need to get into integration. The API failure with timings now has its own ticket and will be followed up on in another PR. The CMS failure appears to be CircleCI only and will need some follow-up investigation. Quality will not pass because it's going to be comparing against master, but once merged in a follow up PR will pass.

So, in summary, this should be ready for merge in order to avoid blocking integration testers anymore and I'll work on one or more follow-up PRs to get the builds properly green.

**Comment by bradenmacdonald** [comment]
@Kelketek I'm seeing this error in the LMS when I try to view the progress page in Apros:
```
2015-08-28 12:20:09,338 ERROR 2571 [django.request] base.py:213 - Internal Server Error: /api/server/users/1/courses/BradenX/101/2015/grades
Traceback (most recent call last):
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/core/handlers/base.py", line 109, in get_response
    response = callback(request, *callback_args, **callback_kwargs)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/rest_framework/compat.py", line 137, in view
    return self.dispatch(request, *args, **kwargs)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/views/decorators/csrf.py", line 77, in wrapped_view
    return view_func(*args, **kwargs)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/rest_framework/views.py", line 400, in dispatch
    response = self.handle_exception(exc)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/rest_framework/views.py", line 397, in dispatch
    response = handler(request, *args, **kwargs)
  File "/edx/app/edxapp/edx-platform/lms/djangoapps/api_manager/users/views.py", line 1014, in get
    progress_summary = json.loads(queryset[0].progress_summary)
  File "/usr/lib/python2.7/json/__init__.py", line 326, in loads
    return _default_decoder.decode(s)
  File "/usr/lib/python2.7/json/decoder.py", line 366, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib/python2.7/json/decoder.py", line 384, in raw_decode
    raise ValueError("No JSON object could be decoded")
ValueError: No JSON object could be decoded
[28/Aug/2015 12:20:09] "GET /api/server/users/1/courses/BradenX/101/2015/grades HTTP/1.0" 500 270994
```

(Could be a configuration error on my part?)

**Comment by bradenmacdonald** [comment]
:+1:

**Comment by smarnach** [comment]
:+1:
